packages/twenty-front/src/modules/favorites/components/CurrentWorkspaceMemberFavorites.tsx

```diff
@@ -1,124 +1,244 @@
+import { useIsMobile } from '@/ui/utilities/responsive/hooks/useIsMobile';
 
 const StyledAvatar = styled(Avatar)`
   :hover {
     cursor: grab;
   }
 `;
 
-const StyledNavigationDrawerItem = styled(NavigationDrawerItem)`
-  :active {
-    cursor: grabbing;
+const StyledDropdownContainer = styled.div`
+  padding: ${({ theme }) => theme.spacing(1)};
+`;
+const StyledIconDotsVertical = styled(IconDotsVertical)<{ isMobile: boolean }>`
+  visibility: ${({ isMobile }) => (isMobile ? 'visible' : 'hidden')};
+  color: ${({ theme }) => theme.color.gray50};
 
-    .fav-avatar:hover {
-      cursor: grabbing;
-    }
+  .navigation-drawer-item:hover &,
+  .navigation-drawer-sub-item:hover & {
+    visibility: visible;
   }
 `;
 
-export const CurrentWorkspaceMemberFavorites = () => {
-  const currentWorkspaceMember = useRecoilValue(currentWorkspaceMemberState);
-
-  const { favorites, handleReorderFavorite } = useFavorites();
-  const loading = useIsPrefetchLoading();
-  const { toggleNavigationSection, isNavigationSectionOpenState } =
-    useNavigationSection('Favorites');
-  const isNavigationSectionOpen = useRecoilValue(isNavigationSectionOpenState);
+const StyledIconHeartOff = styled(IconHeartOff)<{ isMobile: boolean }>`
+  visibility: ${({ isMobile }) => (isMobile ? 'visible' : 'hidden')};
+  color: ${({ theme }) => theme.color.red};
 
-  if (loading && isDefined(currentWorkspaceMember)) {
-    return <FavoritesSkeletonLoader />;
+  .navigation-drawer-sub-item:hover & {
+    visibility: visible;
   }
+`;
+type CurrentWorkspaceMemberFavoritesProps = {
+  folder: {
+    folderId: string;
+    folderName: string;
+    favorites: Array<{
+      id: string;
+      labelIdentifier: string;
+      avatarUrl: string;
+      avatarType: AvatarType;
+      link: string;
+      recordId: string;
+    }>;
+  };
+  isGroup: boolean;
+  handleReorderFavorite: OnDragEndResponder;
+  isOpen: boolean;
+  onToggle: (folderId: string) => void;
+};
 
-  const currentWorkspaceMemberFavorites = favorites.filter(
-    (favorite) => favorite.workspaceMemberId === currentWorkspaceMember?.id,
+export const CurrentWorkspaceMemberFavorites = ({
+  folder,
+  isGroup,
+  handleReorderFavorite,
+  isOpen,
+  onToggle,
+}: CurrentWorkspaceMemberFavoritesProps) => {
+  const currentPath = useLocation().pathname;
+  const theme = useTheme();
+  const isMobile = useIsMobile();
+
+  const [isRenaming, setIsRenaming] = useState(false);
+  const [folderName, setFolderName] = useState(folder.folderName);
+  const { renameFolder, deleteFolder } = useFavoriteFolders();
+  const { closeDropdown } = useDropdown(`favorite-folder-${folder.folderId}`);
+  const selectedFavoriteIndex = folder.favorites.findIndex(
+    (favorite) => favorite.link === currentPath,
   );
+  const { deleteFavorite } = useFavorites();
+
+  const subItemArrayLength = folder.favorites.length;
+
+  const handleSubmitRename = async (value: string) => {
+    const trimmedValue = value.trim();
```

Comment:
we should be allowed to add spaces in a folder name

Reply:
okay!